cql2pgjson/src/test/java/org/folio/cql2pgjson/util/Cql2SqlUtilTest.java

```diff
@@ -153,6 +171,214 @@ public void cql2regexp(String cql, String sql) {
     assertThat(Cql2SqlUtil.cql2regexp(cql), is(sql));
   }
 
+  private String selectTsvector(String field, boolean removeAccents) {
+    return removeAccents ? "SELECT to_tsvector('simple', f_unaccent('" + field.replace("'", "''") + "')) @@ "
+                         : "SELECT to_tsvector('simple', '" + field.replace("'", "''") + "') @@ ";
+  }
+
+  private void assertCql2tsqueryAnd(String field, String query, boolean removeAccents, String result) {
+    try {
+      String sql = selectTsvector(field, removeAccents) + Cql2SqlUtil.cql2tsqueryAnd(query, removeAccents);
+      assertThat(sql, firstColumn(sql), contains(result));
+    } catch (QueryValidationException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void assertCql2tsqueryOr(String field, String query, boolean removeAccents, String result) {
+    try {
+      String sql = selectTsvector(field, removeAccents) + Cql2SqlUtil.cql2tsqueryOr(query, removeAccents);
+      assertThat(sql, firstColumn(sql), contains(result));
+    } catch (QueryValidationException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void assertCql2tsqueryPhrase(String field, String query, boolean removeAccents, String result) {
+    try {
+      String sql = selectTsvector(field, removeAccents) + Cql2SqlUtil.cql2tsqueryPhrase(query, removeAccents);
+      assertThat(sql, firstColumn(sql), contains(result));
+    } catch (QueryValidationException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void assertCql2tsqueryAnd(String field, String query, String result) {
+    assertCql2tsqueryAnd(field, query, false, result);
+  }
+
+  private void assertCql2tsqueryOr(String field, String query, String result) {
+    assertCql2tsqueryOr(field, query, false, result);
+  }
+
+  private void assertCql2tsqueryPhrase(String field, String query, String result) {
+    assertCql2tsqueryPhrase(field, query, false, result);
+  }
+
+  String [] cql2tsqueryParams() {
+    return new String [] {
+        "abc",
+        "abc*",
+        " abc ",
+        "  abc  ",
+        "   abc   ",
+        "abc  xyz",
+        "abc   xyz",
+        "a b c",
+        "abc def ghi",
+        "abc* def* ghi*",  // 'abc':* & 'def':* & 'ghi':*
+                           // minus in words, RMB-438
+        "abc-def",         // 'abc-def' & 'abc' & 'def'
+        "abc--def",        // 'abc' & 'def'
+        "abc---def",       // 'abc' & 'def'
+        "abc-def-ghi",     // 'abc-def-ghi' & 'abc' & 'def' & 'ghi'
+        "abc-def-ghi*",    // 'abc-def-ghi':* & 'abc':* & 'def':* & 'ghi':*
+        "abc-def ghi-jkl", // 'abc-def' & 'abc' & 'def' & 'ghi-jkl' & 'ghi' & 'jkl'
+        "0cc175b9-c0f1-b6a8-31c3-99e269772661",
+        "12345678-9999-0000-1234-012345678901",
```

Comment:
Now that UUIDs are better matched (was & before).. I think it would be nice with a test using
a commutation of those ones.. Eg expect f(alse) for 0cc175b9-c0f1-31c3-b6a8-99e269772661